coreneuron/utils/reports/binary_report_handler.cpp

```diff
@@ -0,0 +1,129 @@
+#include "binary_report_handler.h"
+#include "coreneuron/utils/reports/nrnsection_mapping.h"
+#ifdef ENABLE_REPORTING
+#include "reportinglib/Records.h"
+#endif  // ENABLE_REPORTING
+
+namespace coreneuron {
+
+
+void BinaryReportHandler::create_report(double dt, double tstop, double delay) {
+#ifdef ENABLE_REPORTING
+    records_set_atomic_step(dt);
+#endif  // ENABLE_REPORTING
+    ReportHandler::create_report(dt, tstop, delay);
+}
+
+#ifdef ENABLE_REPORTING
+void BinaryReportHandler::register_soma_report(NrnThread& nt, ReportConfiguration& config,
+                                                VarsToReport& vars_to_report) {
+    int sizemapping = 1;
+    int extramapping = 5;
+    // first column i.e. section numbers
+    int mapping[] = {0};
+    // first row, from 2nd value (skip gid)
+    int extra[] = {1, 0, 0, 0, 0};
+
+    VarsToReport::iterator it;
+    for (it = vars_to_report.begin(); it != vars_to_report.end(); ++it) {
+        int gid = it->first;
+        std::vector<VarWithMapping>& vars = it->second;
+        if (!vars.size())
+            continue;
+        NrnThreadMappingInfo* mapinfo = (NrnThreadMappingInfo*)nt.mapping;
```

Comment:
Indeed, the proper cast should be used instead of a C-style cast, something like:
```cpp
        const auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
```
(Maybe `const` qualifier needs to be removed).